cmd/soroban-rpc/internal/util/monitoredroutine.go

```diff
@@ -0,0 +1,78 @@
+package util
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"runtime"
+	"runtime/debug"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stellar/go/support/log"
+)
+
+type MonitoredRoutineConfiguration struct {
+	Log                *log.Entry
+	LogPanicsToStdErr  bool
+	ExitProcessOnPanic bool
+	PanicsCounter      prometheus.Counter
+}
+
+// MonitoredRoutine give us the ability to spin a goroutine, with clear upfront definitions on what should be done in the
+// case of an internal panic.
+func MonitoredRoutine(cfg MonitoredRoutineConfiguration, fn func()) {
+	go func() {
```

Comment:
This `go` seems somewhat unnecessary, given there's nothing inherently async here. This is just a fancy try-catch block. So, if the user wants they could easily do:
```
go MonitoredRoutine(cfg, func() { })
```
(This would also make it compatible with waitgroups/errgroups/other-concurrency-mechanisms)